sklearn/utils/estimator_checks.py

```diff
@@ -1177,6 +1184,21 @@ def check_estimators_unfitted(name, estimator_orig):
 
     msg = "fit"
     if hasattr(est, 'predict'):
+        can_predict = False
+        try:
+            # some models can predict without fitting
+            # like GaussianProcess regressors
+            # in this case, we skip this test
+            pred = est.predict(X)
+            assert_equal(pred.shape[0], X.shape[0])
+            can_predict = True
+        except:
+            pass
+        if can_predict:
+            raise SkipTest(
+                "{} can predict without fitting, skipping "
+                "check_estimator_unfitted.".format(name))
+
```

Comment:
Why? The test is that if it fails, it fails with a helpful message. That's what the test was written for afaik. So now we allow to also pass. If it fails, it still needs to have a helpful message.